src/redux/offersSlice.js

```diff
@@ -0,0 +1,52 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { getOffers } from 'api';
+
+const initialState = {
+  offers: [],
+  pauseOffers: [],
+  pauseOfferIDs: [],
+  loading: false,
+  error: null
+};
+
+export const fetchOffers = createAsyncThunk(
+  'offers/fetchOffers',
+  async thunkAPI => {
+    try {
+      const result = await getOffers();
+      return result;
+    } catch (err) {
+      return thunkAPI.rejectWithValue(err.message);
```

Comment:
```
 async ({rejectWithValue}) => {
    try {
      const result = await getOffers();
      return result;
    } catch (err) {
      return rejectWithValue(err.message);
```

Reply:
This one is not working but if we want this convention we need to do something like this (this first argument is required)
`async (arg, { rejectWithValue }) => {
    try {
      const result = await getOffers();
      return result;
    } catch (err) {
      return rejectWithValue(err.message);
    }
  }`

Reply:
true :)